src/Providers/HttpService.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Net;
+using System.IO;
+using System.Xml.Linq;
+using System.Runtime.Serialization.Json;
+using System.Threading.Tasks;
+using System.Text;
+
+namespace clio.Providers
+{
+	public class HttpService
+	{
+		internal HttpService ()
+		{
+		}
+
+		public static string UserAgent = "Xamarin";
+
+		public async Task<string> PostAsync (string url, string content = null, string user = null, string pwd = null, string contentType = null, int? expectedStatus = null, string method = "POST", string accept = null)
+		{
+			using (var sr = await PostAsStreamAsync (url, content, user, pwd, contentType, expectedStatus, method))
+			{
+				return new StreamReader (sr).ReadToEnd ();
+			}
+		}
+
+		public async Task<Stream> PostAsStreamAsync (string url, string content = null, string user = null, string pwd = null, string contentType = null, int? expectedStatus = null, string method = null, string accept = null)
+		{
+			try
+			{
+				HttpWebRequest req = (HttpWebRequest)WebRequest.Create (url);
+				req.UserAgent = HttpService.UserAgent;
+				req.SendChunked = false;
+				req.KeepAlive = false;
+				req.Expect = "";
+				req.AllowWriteStreamBuffering = true;
+				req.Method = method ?? "POST";
+				req.Headers.Add ("Content-Encoding", "utf-8");
+				if (accept != null)
+					req.Accept = accept;
+				if (contentType != null)
+					req.ContentType = contentType;
+				//				if (user != null)
```

Comment:
Commented out code?

Reply:
that's what you get when you yank code from someplace else. will tidy up.